backend/app/api/api_v1/endpoints/domains.py

```
@@ -6352,16 +6354,36 @@ def _reputation_recommendations(item: Optional[SourceReputation]) -> List[Source
         return None
-    try:
-        return await asyncio.wait_for(provider.lookup_ptr(ip), timeout=timeout)
-    except Exception:
-        return None
+    for candidate in _ptr_lookup_providers(provider):
+        try:
+            hostname = await asyncio.wait_for(candidate.lookup_ptr(ip), timeout=timeout)
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            continue
```

Ruff (0.15.20)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)